addons/mrp/models/mrp_production.py

```diff
@@ -1207,9 +1211,13 @@ def action_confirm(self):
                         'product_uom_qty': move_finish.product_uom._compute_quantity(move_finish.product_uom_qty, move_finish.product_id.uom_id),
                         'product_uom': move_finish.product_id.uom_id
                     })
-            production.move_raw_ids._adjust_procure_method()
-            (production.move_raw_ids | production.move_finished_ids)._action_confirm(merge=False)
-            production.workorder_ids._action_confirm()
+            production.write(production_vals)
+            move_raws_to_adjust |= production.move_raw_ids
+            moves_to_confirm |= (production.move_raw_ids | production.move_finished_ids)
+            workorder_to_confirm |= production.workorder_ids
```

Comment:
It's even faster (especially in case of large number of records) to fill a `set()` with the ids then access the records via 
`self.env[model].browse(ids_set)`